src/mail-app/mail/view/LabelsPopup.ts

```diff
@@ -114,7 +124,34 @@ export class LabelsPopup implements ModalComponent {
 		}
 	}
 
-	private applyLabels() {
+	private isMaxLabelsReached(): boolean {
+		const { addedLabels, removedLabels } = this.getSortedLabels()
```

Comment:
it seems a bit wasteful to get label states twice. In `getLabelStatesForMails()` we already count labels which is what we want anyway. Could we reuse that information somehow?

If not, no big deal, I don't think we will have any issues with it.